### PR TITLE
Legacy find/replace handling for emails.

### LINKED ISSUE #16705: Email template variables broken in latest master
# Explanation
It seems we're now replacing the keys and values of `$this->placeholders` only in case the filters `woocommerce_email_format_string_replace` or `woocommerce_email_format_string_find` are in use.

# Steps to reproduce
1. Order any item
2. Check order receipt e-mail
Expected: Variables are populated
Actual: Variables are not populated

# Screenshot
screen shot 2017-09-05 at 08 20 15 am

# Investigation
```
21bd0e539780d9489c8a57c7048ba8a45c07a634 is the first bad commit
commit 21bd0e539780d9489c8a57c7048ba8a45c07a634
Author: Mike Jolley <mike.jolley@me.com>
Date:   Thu Aug 31 12:21:02 2017 +0100

    Better handling for legacy email find/replace filters
    
    Closes #16659

:040000 040000 5264a36dbefa48fb194206eaf1a798683be6186f 29aa4113c73dc35cb4d8d226b92a9573e68421d7 M	includes
```

### PR DESCRIPTION
**Pull request by mikejolley**
Closes #16705

### CODE REVIEW

File: includes/emails/class-wc-email.php [hunk 1]
```diff
@@ -250,14 +250,23 @@ public function handle_multipart( $mailer ) {
...
+	 * @param mixed $string Text to replace placeholders in.
 	 * @return string
 	 */
 	public function format_string( $string ) {
-		// Legacy placeholders. @todo deprecate in 4.0.0.
+		$find    = array_keys( $this->placeholders );
+		$replace = array_values( $this->placeholders );
+
+		// If using legacy find replace, add those to our find/replace arrays first. @todo deprecate in 4.0.0.
+		if ( isset( $this->find, $this->replace ) ) {
```
Comment (bor0) [inline includes/emails/class-wc-email.php]: `isset` on an empty array (which is initialized by default in this class) will return true. Do we want `! empty`?

File: includes/emails/class-wc-email.php [hunk 2]
```diff
@@ -250,14 +250,23 @@ public function handle_multipart( $mailer ) {
...
+			$find    = array_merge( $this->find, $find );
+			$replace = array_merge( $this->replace, $replace );
+		}
+
+		// If using the older style filters for find and replace, ensure the array is associative and then pass through filters. @todo deprecate in 4.0.0.
 		if ( has_filter( 'woocommerce_email_format_string_replace' ) || has_filter( 'woocommerce_email_format_string_find' ) ) {
-			$legacy_find    = array();
-			$legacy_replace = array();
+			$legacy_find    = isset( $this->find ) ? $this->find : array();
```
Comment (bor0) [inline includes/emails/class-wc-email.php]: Can drop the `isset` here, as `find`/`replace` are initialied to empty array in this class.

File: includes/emails/class-wc-email.php (hunk 1 above)
Comment (mikejolley) [reply includes/emails/class-wc-email.php]: Would it hurt to remove the variables from the class?

Comment (bor0) [reply includes/emails/class-wc-email.php]: Probably not. I am thinking if we should just use `in_array` instead. Can legacy `find`/`replace` be strings?

Comment (mikejolley) [reply includes/emails/class-wc-email.php]: Arrays only. Core no longer sets them. If a plugin extending woocommerce used them, they would be set, so if we can remove them from the class, great.

Comment (bor0) [reply includes/emails/class-wc-email.php]: Then in my opinion we can leave the `isset`s and remove the `find`/`replace` definitions of this class 👍

**Review by bor0 — APPROVED** [review]

**Pull request merged**